Makefile

```diff
@@ -179,11 +181,17 @@ endif
 # Generate bundle manifests and metadata, then validate generated files.
 .PHONY: bundle
 bundle: manifests kustomize
+ifneq ($(and $(REPLACES),$(OPERATOR_NAMESPACE)),)
 	operator-sdk generate kustomize manifests -q
 	cd config/manager && $(KUSTOMIZE) edit set image controller=$(IMG)
+	cd config/manifests/bases && sed -e "s/@REPLACES@/$(REPLACES)/" -e "s/@VERSION@/$(VERSION)/" kustomization-base.yaml > kustomization.yaml
```

Comment:
@leelavg I think this change came from a misunderstanding of the intant of my prev request. 
I would prefer to edit the actual file and not create a template to be edited by `sed`. Please use `yq` or any other tool (even `kustomize` itself if you find it easier) to edit the actual patch file.

Reply:
- Let me explain why I did what I did and I don't want to use neither `sed` nor  introduce a new one `yq`

> 	cd config/manifests/bases && \
- normal cd op

>		rm -rf kustomization.yaml && \
- we can only update below resources/fields with `kustomize edit set`
```
image
label
nameprefix
namespace
namesuffix
replicas
```
- other op `kustomize edit add` will add to the existing resource patch file and since we can't override existing field in the patch file, we are removing it

>		$(KUSTOMIZE) create --resources $(BUNDLE_PACKAGE).clusterserviceversion.yaml && \
- this creates a patch file with resource section `$(BUNDLE_PACKAGE).clusterserviceversion.yaml` on which patches has to be applied not the CSV itself

>		$(KUSTOMIZE) edit add annotation --force 'olm.skipRange':">=0.0.1 <$(VERSION)" && \
- first change we want in a csv, i.e, to update the skipRange

>		$(KUSTOMIZE) edit add patch --name $(BUNDLE_PACKAGE).v0.0.0 --kind ClusterServiceVersion \
		--patch '[{"op": "replace", "path": "/spec/replaces", "value": "$(REPLACES)"}]'
- unlike `kustomize edit add annotation` we can't directly target the fields under spec so we are adding a strategic merge patchable json object

---
- Against sed:
1. If we want inplace replace, the resulting expression will be cryptic and may not be cross-platform
- Against yq: 
1. extra dependency, the field which we want to edit is a json obj inside string and it'll more or less result in same format as kustomize minus 2 steps (remove and create of kustomization.yaml)
2. those two steps may not cause us much friction while generating bundle and can get away with one less installable dependency

---
- All I'm doing is to clarify as much as possible and correct any of my misunderstandings how the tools work and I believe these changes doesn't need a lot of discussions